src/actions/measurements.ts

```diff
@@ -517,14 +577,176 @@ export const changeMeasurementsDisplay = (
 export const changeMeasurementsGroupBy = (
   newGroupBy: string
 ): ThunkFunction => (dispatch, getState) => {
-  const { measurements } = getState();
-  const controlKey = "measurementsGroupBy";
-  const newControls = { [controlKey]: newGroupBy };
+  const { controls, measurements } = getState();
+  const groupingValues = measurements.collectionToDisplay.groupings.get(newGroupBy).values || [];
+  const newControls: Partial<MeasurementsControlState> = {
+    /* If the measurementsColorGrouping is no longer valid, then set to undefined */
+    measurementsColorGrouping: groupingValues.includes(controls.measurementsColorGrouping)
+      ? controls.measurementsColorGrouping
+      : undefined,
+    measurementsGroupBy: newGroupBy
+  };
 
-  dispatch({
-    type: CHANGE_MEASUREMENTS_GROUP_BY,
-    controls: newControls,
-    queryParams: createMeasurementsQueryFromControls(newControls, measurements.collectionToDisplay, measurements.defaultCollectionKey)
+  batch(() => {
+    dispatch({
+      type: CHANGE_MEASUREMENTS_GROUP_BY,
+      controls: newControls,
+      queryParams: createMeasurementsQueryFromControls(newControls, measurements.collectionToDisplay, measurements.defaultCollectionKey)
+    });
+
+    /* After the group by has been updated, update the measurement coloring data if needed */
+    updateMeasurementsColorData(
+      newControls.measurementsColorGrouping,
+      controls.measurementsColorGrouping,
+      controls.colorBy,
+      controls.defaults.colorBy,
+      dispatch
+    );
+  })
+}
+
+export function getActiveMeasurementFilters(
+  filters: MeasurementFilters
+): {string?: string[]} {
+  // Find active filters to filter measurements
+  const activeFilters: {string?: string[]} = {};
```

Comment:
What does making the key part of this object optional accomplish? IOW, how is this different from `{string: string[]}` as a type declaration?

Reply:
The optional key allows `activeFilters` to be an empty object, which is needed since there are cases where all of the filters are "inactive".

Reply:
ah, okay, thanks -- I guess in the past I've spelled this 

```
const foo: Record<string,string[]> = {};
```

which doesn't seem to have the same problem with an empty object. (Not asking for a change.)